internal/app/admin/http/handler.go

```diff
@@ -102,45 +102,62 @@ func configDumpHandler(bootstrapConfig *bootstrapv1.Bootstrap) http.HandlerFunc
 	}
 }
 
-// TODO(lisalu): Support dump of matching resources when cache key regex is provided.
+func printCacheEntry(key string, cache cache.ReadOnlyCache, w http.ResponseWriter) {
+	resource, err := cache.FetchReadOnly(key)
+	if err != nil {
+		fmt.Fprintf(w, "no resource for key %s found in cache.\n", key)
+		return
+	}
+	resourceString, err := resourceToString(resource)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "unable to convert resource to string.\n")
+		return
+	}
+	fmt.Fprintf(w, "%s: %s\n", key, resourceString)
+}
+
 func cacheDumpHandler(o *orchestrator.Orchestrator) http.HandlerFunc {
 	return func(w http.ResponseWriter, req *http.Request) {
 		cacheKey := getParam(req.URL.Path)
 		cache := orchestrator.Orchestrator.GetReadOnlyCache(*o)
 
-		// If no key is provided, output the entire cache.
-		if cacheKey == "" {
+		// If wildcard suffix provided, output all cache entries that match given prefix. If no key is
+		// provided, output the entire cache.
+		containsWildcardSuffix := strings.HasSuffix(cacheKey, "*")
+		if cacheKey == "" || containsWildcardSuffix {
+			rootCacheKeyName := strings.TrimSuffix(cacheKey, "*")
+
+			// Retrieve all keys
 			keys, err := orchestrator.Orchestrator.GetDownstreamAggregatedKeys(*o)
 			if err != nil {
 				w.WriteHeader(http.StatusInternalServerError)
 				fmt.Fprintf(w, "error in getting cache keys: %s", err.Error())
 				return
 			}
-			for key := range keys {
-				resource, err := cache.FetchReadOnly(key)
-				if err == nil {
-					resourceString, err := resourceToString(resource)
-					if err == nil {
-						fmt.Fprintf(w, "%s: %s\n", key, resourceString)
+
+			// Find keys that match prefix of wildcard
+			if containsWildcardSuffix {
+				for potentialMatchKey := range keys {
+					if !strings.HasPrefix(potentialMatchKey, rootCacheKeyName) {
+						delete(keys, potentialMatchKey)
```

Comment:
is it safe to delete from a list/map while iterating on it?

Reply:
I also was concerned about this, but I think it is safe. Golang's [docs](https://golang.org/ref/spec#For_statements) imply that it is okay to delete entries while iterating through a map in rule three of the "For statements with range clause" section. Let me know if I misinterpreted it though!

Reply:
it's a very unusual design decision, but yeah, it totally works.